### PR TITLE
CPM-1151: Add progress indicator and additional fields in attribute c…

### PR DESCRIPTION
**Pull request by cemorin**
…reation modal

**Description (for Contributor and Core Developer)**

Progress indicator is not the correct one, waiting for this PR:
https://github.com/akeneo/pim-community-dev/pull/20002

Screenshot from 2023-07-11 16-21-53

Screenshot from 2023-07-11 16-21-45

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by antoinearrive — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (selectedType === 'pim_catalog_table')
+      commonSteps.splice(1, 0, {
+        current: currentStepIndex === 1,
+        label: 'pim_enrich.entity.attribute.property.attribute_creation_template',
+      });
+
+    return commonSteps;
+  }, [selectedType, currentStepIndex]);
+
+  console.log({steps});
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx]: to remove

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+type Props = {
+  currentStepIndex: number;
+  selectedType?: string;
+};
+
+const CreateAttributeProgressIndicator: React.FC<Props> = ({currentStepIndex, selectedType}) => {
+  const translate = useTranslate();
+  const isSettingsStep = selectedType === 'pim_catalog_table' ? currentStepIndex === 2 : currentStepIndex === 1;
+  console.log({currentStepIndex});
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx]: to remove

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx [hunk 3]
```diff
@@ -87,6 +113,37 @@ const CreateAttributeModal: React.FC<CreateAttributeModalProps> = ({
         {extraFields.map((field: CreateAttributeModalExtraField, i: number) =>
           React.cloneElement<CreateAttributeModalExtraFieldProps>(field.component, {key: i})
         )}
+        <CheckBoxesContainer>
+          <Checkbox
+            readOnly={initialData?.attribute_type === 'pim_catalog_identifier'}
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx:118]: should be readonly for every attribute type but 'pim_catalog_text' and 'pim_catalog_number'

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx [hunk 4]
```diff
@@ -87,6 +113,37 @@ const CreateAttributeModal: React.FC<CreateAttributeModalProps> = ({
...
+          >
+            {translate('pim_enrich.entity.attribute.property.unique')}
+          </Checkbox>
+          <Tooltip direction="top">
+            <b>{translate('pim_enrich.entity.attribute.property.unique')}</b>
+            <p>{translate('pim_enrich.entity.attribute.property.unique_value_helper')}</p>
+          </Tooltip>
+        </CheckBoxesContainer>
+        <CheckBoxesContainer>
+          <Checkbox checked={isScopable} onChange={setIsScopable}>
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx:130]: should be readonly for pim_catalog_identifier

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx [hunk 5]
```diff
@@ -87,6 +113,37 @@ const CreateAttributeModal: React.FC<CreateAttributeModalProps> = ({
...
+            {translate('pim_enrich.entity.attribute.property.scopable')}
+          </Checkbox>
+          <Tooltip direction="top">
+            <b>{translate('pim_enrich.entity.attribute.property.scopable')}</b>
+            <p>{translate('pim_enrich.entity.attribute.property.scopable_helper')}</p>
+          </Tooltip>
+        </CheckBoxesContainer>
+        <CheckBoxesContainer>
+          <Checkbox checked={isLocalizable} onChange={setIsLocalizable}>
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx:139]: same

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/attribute/form/CreateAttributeButtonApp.tsx [hunk 6]
```diff
@@ -36,8 +38,8 @@ const CreateAttributeButtonApp: React.FC<CreateAttributeButtonAppProps> = ({
   const [isOpen, open, close] = useBooleanState(isModalOpen);
   const [attributeData, setAttributeData] = React.useState<AttributeData>(initialData);
   const [currentStepIndex, setCurrentStepIndex] = React.useState<number>(-1);
-
-  let stepsForAttributeType = steps['default'];
+  console.log('CreateAttributeButtonApp');
```
Comment (mmetayer) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/attribute/form/CreateAttributeButtonApp.tsx:41]: to remove

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    const commonSteps = [
+      {current: currentStepIndex === -1, label: 'pim_enrich.entity.attribute.property.attribute_creation_type'},
+      {current: isSettingsStep, label: 'pim_enrich.entity.attribute.property.attribute_creation_settings'},
+    ];
+
+    if (selectedType === 'pim_catalog_table')
+      commonSteps.splice(1, 0, {
+        current: currentStepIndex === 1,
+        label: 'pim_enrich.entity.attribute.property.attribute_creation_template',
+      });
```
Comment (mmetayer) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx:30]: I suppose there is not a more elegant way to do that?

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx (hunk 3 above)
Comment (cemorin) [reply src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/CreateAttributeModal.tsx:118]: Ahhhh yes!!!

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx (hunk 7 above)
Comment (cemorin) [reply src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/CreateAttributeProgressIndicator.tsx]: I tried :angel:

**Review by Beninho — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Resources/translations/jsmessages.en_US.yml [hunk 8]
```diff
@@ -223,6 +223,14 @@ pim_enrich.entity.attribute:
             invalid: The code is invalid
             not_available: This code is not available
             is_duplicate: This code is already used
+        identifiers_limit: You can create a maximum of 10 identifiers. Learn more about multiple identifiers in the
+        identifiers_limit_link: help center.
+        unique_value_helper: Should the value of this attribute be unique? You will not be able to change this setting later.
+        scopable_helper: Should the value of this attribute be unique for each channel (scopable)?
+        localizable_helper: Should the value of this attribute be unique for each locale (localizable)?
```
Comment (pierallard) [inline src/Akeneo/Pim/Structure/Bundle/Resources/translations/jsmessages.en_US.yml:230]: Not fan of these sentences.
"Should the value of this attribute be set for each channel", not "unique", IMHO.

**Review by pierallard — APPROVED** [review]

**Pull request merged**